src/AuthContext.tsx

```diff
@@ -0,0 +1,54 @@
+import type { ReactNode } from 'react';
+import { createContext, useContext, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+type AuthContextType = {
+  isAuthenticated: boolean;
+  username: string | null;
+  login: (user: string) => void;
+  logout: () => void;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isAuthenticated, setIsAuthenticated] = useState(false);
```

타입이 추론될 수 있는 경우엔 굳이 명시적으로 남기지 않아도 좋아요!